src/jdk.jfr/share/classes/jdk/jfr/internal/EventInstrumentation.java

```diff
@@ -676,7 +676,7 @@ void updateInstanceCommit(BlockCodeBuilder blockCodeBuilder, Label end, Label ex
         // stack: [EW] [EW]
         getEventConfiguration(blockCodeBuilder);
         // stack: [EW] [EW] [EC]
-        blockCodeBuilder.loadConstant(Opcode.LDC2_W, eventTypeId);
+        blockCodeBuilder.loadConstant(eventTypeId);
```

Comment:
Does this means an int (single slot) can be pushed if the event type id is below Integer.MAX_VALUE?

Reply:
No, because the type of `eventTypeId` is `long`, so the boxed type is `Long`, therefore the only instructions are `LDC2_W`, `LCONST_0`, and `LCONST_1`:

https://github.com/openjdk/jdk/blob/86c4104d86d91fefae3d9dc11e4f2684d29fcde1/src/java.base/share/classes/java/lang/classfile/CodeBuilder.java#L639-L642

Reply:
No, because int and long are not interchangeable in the jvm. Only subints can be replaced by ints.